management-core/src/main/java/com/ns/task/repositories/ProductRepositoryImpl.java

```diff
@@ -21,12 +22,13 @@
     }
 
     @Override
-    public ProductEntity saveProduct(ProductEntity product) {
+    public ProductEntity saveProduct(ProductEntity product)  throws SQLIntegrityConstraintViolationException {
         StoredProcedureQuery insertion = manager.createNamedStoredProcedureQuery("insertProduct");
         insertion.setParameter("productName", product.getName());
         insertion.setParameter("model", product.getModel());
         insertion.setParameter("price", product.getPrice());
-        insertion.execute();
+        boolean resultForInsertion = insertion.execute();
```

Comment:
What happens if `insertion.execute()` is false?

`resultForInsertion` is never used

Reply:
According to the documentation Return true if the first result corresponds to a result set,
	 * and false if it is an update count or if there are no results
	 * other than through INOUT and OUT parameters, if any.
So in my case always will return false because I do not return anything